---
title: "Ly_etal_PMA_ozonated_wastewater"
author: "Pieter van Veelen"
date: "2024-03-25"
output: html_document
---

This analysis workflow covers the data analysis and results of three experiments reported by Ly et al. 2024, in which the efficacy of propidium monoazide (PMA) pre-treatment on membrane filtered water samples is validated for the quantitative assessment of bacterial survival using qPCR. The method is applied to Gram positive and negative species (Exp. I), at different starting concentrations of *E. coli* (Exp. II), and tested in an advanced wastewater treatment setting via wastewater treatment plant effluent treated in an ozonation reactor. Bacterial survival estimates were compared for PMA-qPCR, conventional qPCR (tested as a blank PBS control pre-treatment without PMA) and with culture based determination of bacterial abundances (Exp. III) of susceptible *E. coli* and relatively ozone resistant *P. aeruginosa* spiked to effluent prior to treatment.


```{r setup, include=FALSE}

### Workflow setup
knitr::opts_chunk$set(echo = TRUE)
options(scipen = 999)
options(digits = 3)

if(!dir.exists("input_data")){dir.create("input_data")}
if(!dir.exists("output_data")){dir.create("output_data")}
if(!dir.exists("figures")){dir.create("figures")}
if(!dir.exists("scripts")){dir.create("scripts")}

```

```{r install and load packages, warning=F, message=F, echo=T, eval=T, include=F}

# list required packages
packages = c("scales","ggtext","ggrepel","phia","openxlsx","lme4","lmerTest","modelr","insight","effects","broom.mixed","easystats","broom","emmeans","rstatix","wesanderson","scales","lemon","tidyverse")

# Loop through each package name and install when not installed
for (pkg in packages) {
  # Check if the package is already installed
  if (!requireNamespace(pkg, quietly = TRUE)) {
    # If not installed, install the package
    install.packages(pkg)
  }
  # Load the package
  library(pkg, character.only = TRUE)
}

```

```{r define colors}

# color pallette
#pal = wes_palette("FantasticFox1", n = 5)
#show_col("#8ACDEA", "#FCB97D")
#show_col("#2E86AB","#F56476")
pal = c("#8ACDEA", "#FCB97D","#2AB7CA","grey10","#F3B61F", "#2E86AB","#F56476")


```

### Data import and reshape

```{r import data, warning=F, message=F, echo=F, eval=T, include=T}

### Experiment I ###

## Testing PMA pre-treatment efficacy with different life/dead proportions of 4 bacterial species, 2 type strains G+ and G-, and 2 isolates from ozonated wastewater

# read qPCR data for Exp. I
data = openxlsx::read.xlsx("/Users/pvee/OneDrive - Wetsus/General - PMA treatment in molecular analyses/qPCR/PMA_data_qPCR.xlsx") %>% as_tibble()
data = 
  data %>% mutate(exp_id = str_sub(exp_date, start = 12),
                  date = lubridate::ymd(str_sub(exp_date, start = 1, end = 10)))

# check overview sample groups
# data %>% 
#   filter(grepl("Ecoli|Pseudomonas|Acinetobacter|Efaecalis", sample_type)) %>% 
#   group_by(exp_date, sample_type, live_dead, conc, pma, exp_id) %>% count %>% view

# calculate mean of technical duplicates
data_clean =
  data %>% 
  #slice(1:40) %>% 
  filter(grepl("Ecoli|Pseudomonas|Acinetobacter|Efaecalis", sample_type),
         !grepl("2022_07_12|2022_08_10", exp_date)) %>% # remove pilot experiment data with different protocol
  mutate(species = case_when(sample_type == "Ecoli_culture" ~ "*Escherichia coli*",
                             sample_type == "Pseudomonas_culture" ~ "*Pseudomonas citronellolis*",
                             sample_type == "Acinetobacter_culture" ~ "*Stenotrophomonas rhizophila*",
                              sample_type == "Efaecalis_culture" ~ "*Enterococcus faecalis*")) %>% 
  group_by(exp_date, live_dead, conc, pma, species, date) %>% 
  summarise(sq_calc_mean = mean(sq_calc), .groups = "drop") %>% 
  mutate_if(is.character, as.factor) %>% 
  mutate(date = as.factor(date)) %>% 
  mutate(log_sq_calc_mean = log(sq_calc_mean, 10))

# import data for different starting cell concentrations
data_exp1 = openxlsx::read.xlsx("~/OneDrive - Wetsus/General - PMA treatment in molecular analyses/internships/Nick Middelbos/Resultaten/qPCR/final_databases/NMID_Exp1_Cell_density_effect_Ecoli_KOSIdata_added_v2.xlsx") %>% 
  as_tibble(rownames = "row.names") %>% 
  rename(conc = prop_life, pma = PMA, sq_calc = copies_ml) %>% 
  mutate(pma = if_else(pma == "no_PMA", "no PMA", pma),
         exp_id = factor(exp_id, 
                         levels = c("E.coli_low", "E.coli_medium", "E.coli_high"))) 



### Experiment II #####

## Application of PMA pretreatment to spiked ozonated wastewater effluent

# import operation data of ozone reactor
reactor_data = 
  openxlsx::read.xlsx("input_data/raw_qPCR_data/ozone_PMA_reactor_data.xlsx")

# import bacterial concentration data based on culturing
culture_data = 
  openxlsx::read.xlsx("input_data/raw_qPCR_data/ozone_PMA_culturing_data.xlsx")

# import experimental metadat: dates and filtering volumes
qPCR_overview = 
  openxlsx::read.xlsx("input_data/raw_qPCR_data/Overview_exp_PMA.xlsx")

# import (PMA-)qPCR data for 16S rRNA gene amplicons (primers 338F/518R; Fierer et al. 2005)
qPCR_16S_data = 
  openxlsx::read.xlsx("input_data/raw_qPCR_data/qPCR_data_16S.xlsx")%>%
  mutate(qPCR_date = gsub("_", "", qPCR_date),
         Exp_date = gsub("_", "", Exp_date),
         primer = "bac_16S") %>% 
  dplyr::select(-sq_theoretical_16S, -sq_measured_16S, -sq_calc_16S)

# import (PMA-)qPCR data for E.coli (primers ybbw gene; Walker et al. 2017)
qPCR_EC_data = openxlsx::read.xlsx("input_data/raw_qPCR_data/qPCR_data_EC.xlsx") %>%
  mutate(qPCR_date = gsub("_", "", qPCR_date),
         Exp_date = gsub("_", "", Exp_date),
         primer = "ybbw")

# import (PMA-)qPCR data for Pseudomonas aeruginosa (in house developed primers; see SI and below)
# 182F: CGTCCTGAGGGAGAAAGTG
# 582R: CCAACTTGCTGAACCACCTA
qPCR_PA_data = openxlsx::read.xlsx("input_data/raw_qPCR_data/qPCR_data_PA.xlsx")%>%
  mutate(qPCR_date = gsub("_", "", qPCR_date),
         Exp_date = gsub("_", "", Exp_date),
         primer = "PA_marta")

## combine all qPCR data
qPCR_final_data <- 
  bind_rows(qPCR_16S_data, qPCR_EC_data, qPCR_PA_data) %>%
  mutate(Log.Starting.Quantity = as.numeric(Log.Starting.Quantity),
  Sample = ifelse(Content == "NTC-01", "Blank", Sample))


```

```{r ExpII_extract_high_E.coli, warning=F, message=F}

# Extract E.coli high concentration data from Exp. I
high_conc_ecoli_data = 
data_clean %>% 
  dplyr::select(date = exp_date, species, conc, pma, sq_calc_mean) %>% 
  filter(!grepl("2022_07_12|2022_08_10", date),
         species == "*Escherichia coli*") %>% 
  mutate(conc = ifelse(conc == 0, 0.01, conc),
         log_conc = log10(conc),
         log_sq_calc_mean = log10(sq_calc_mean)) %>% 
  mutate(exp_id = "E.coli_high",
         pma = as.character(pma))


# Reshape data maipulated starting concentrations
data_clean3 = 
data_exp1 %>%
  group_by(pma, conc, exp_id, date) %>% 
  reframe(sq_calc_mean = mean(sq_calc)*DNA_diluted, sq_calc_sd = sd(sq_calc),
            ct_calc_mean = mean(Ct)*DNA_diluted, ct_calc_sd = sd(Ct)) %>% 
    mutate(conc = ifelse(conc == 0, 0.01, conc),
         log_conc = log10(conc),
         log_sq_calc_mean = log10(sq_calc_mean),
         date= as.factor(date)) 
      
# Combine Ecoli high concentration data with manipulated starting concentrations
data_clean4 =
  bind_rows(
  high_conc_ecoli_data,
  data_clean3 %>% 
    filter(!grepl("high",exp_id)) %>% 
    dplyr::select(date, pma, conc, exp_id, log_sq_calc_mean)
  ) %>% 
    mutate(pma = factor(if_else(pma == "no PMA", "none", pma)))

```

```{r reshape culturing data, warning=F, message=F}

# filter culturing data on usable data (within counting range) and calculate mean CFU/mL:
culture_data_countable = 
  culture_data %>% 
  filter(countable == 'yes')

# create data frame of CFU concentrations
culture_data_countable_df = 
  culture_data_countable %>% 
  group_by(exp_date, sample_type) %>%
  summarise(mean_CFU_mL = mean(as.numeric(CFU_mL)),
            mean_CFU_100mL = mean_CFU_mL*100,
            .groups = "drop")

# combine data and filter for final analysis
culture_data <- 
  left_join(culture_data_countable, 
            culture_data_countable_df %>% dplyr::select(exp_date:mean_CFU_100mL), 
            by = c('exp_date', 'sample_type'))  %>%
  filter(!duplicated(mean_CFU_100mL))

```

### Ozone reactor run operational diagnostics
The calculation of average measured COD and O3 dose:
```{r calculate ozone dose, warning=F, message=F}

# combine reactor data and culturing data 
complete_culturing_df <- 
  left_join(reactor_data, 
            culture_data %>% dplyr::select(exp_series:mean_CFU_mL), 
            by = c('exp_series', 'person','exp_id','exp_date','species', 'wetsus_strain'))

# ozone doses for six experiments
ozone_dose = 
complete_culturing_df %>% 
  #filter(grepl("effluent", sample_type)) %>% 
  dplyr::select(exp_id, species, sample_type, exp_date, COD_effluent_mg_L, COD_spiked_effluent_mg_L, ozone_COD_g_g) 

# calculate average ozone dose across 6 experiments
ozone_dose %>% 
  group_by(sample_type) %>% 
  summarise(O3_dose_g_gCOD = mean(ozone_COD_g_g), 
            sd_O3 = sd(ozone_COD_g_g), 
            mean_COD_eff_mgL = mean(COD_effluent_mg_L), 
            sd_COD_eff_mgL = sd(COD_effluent_mg_L), 
            mean_COD_spiked_mgL = mean(COD_spiked_effluent_mg_L), 
            sd_COD_spiked_mgL = sd(COD_spiked_effluent_mg_L), 
            n = n())

```

### Results

#### Experiment I: PMA efficacy on four bacterial species as function of life cell concentration

```{r ExpI_LMM model, warning=F, echo=T, eval=T,include=T, message=F}

# a linear mixed effects model is run to estimate the interaction between life-dead ratio (conc) and pma application
# differences in starting concentrations of cultures grown overnight on different dates for replicate experiments are considered random variance
# differences in starting concentrations among species are evaluated as fixed factor as a threeway to assess consistency of PMA efficacy across species including Gram+ and Gram- species

# add 0.01 to zero values for log-log linear analysis

data_lm_expI =
data_clean %>% 
  mutate(conc = ifelse(conc == 0, 0.01, conc),
         log_conc = log10(conc),
         log_sq_calc_mean = log10(sq_calc_mean))


###### LMM with species in three-way interaction #############
# run LMM
model_test_lmer = lmerTest::lmer(log_sq_calc_mean ~  log_conc*pma*species+(1|date), data = data_lm_expI)

# view ANOVA
anova_lmm_test =
anova(model_test_lmer, ddf = "Kenward-Roger") %>% 
  broom::tidy() %>% 
  rename("F.statistic" = statistic) %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     TRUE ~ ""))

# view anova
anova_lmm_test


###### species factor not significantly interacting #############
# F(3,101) = 2.15, P = 0.0989
###### --> update LMM with species as fixed factor #############

# run LMM
model_lmer = lmerTest::lmer(log_sq_calc_mean ~  log_conc*pma+species+(1|date), data = data_lm_expI)

# view ANOVA
anova_lmm =
anova(model_lmer) %>% 
  broom::tidy() %>% 
  rename("F.statistic" = statistic) %>% 
  mutate(
    p.value.format = format(p.value, scientific=T),
    sig = case_when(p.value < 0.0001 ~ "****",
                    p.value < 0.001 ~ "***",
                    p.value < 0.01 ~ "**",
                    p.value < 0.05 ~ "*",
                    TRUE ~ ""))

# view anova
anova_lmm

# save anova results
anova_lmm %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv("output_data/ELLY_DPOL_ExpI_general_LMM_anova_results.csv")

# Inspect the model diagnostic metrics
model.metrics <- augment(model_lmer) #%>%
#dplyr::select(-.hat, -.sigma, -.cooksd) # Remove details
#head(model.metrics)

```
A linear mixed model tested the null hypothesis that qPCR based concentration estimates with and without PMA pre-treatment do not differentially relate to the proportion of live cells in a cell suspension. The alternative hypothesis is a positive relationship with the proportion of live cells in PMA-pretreated cell suspension samples, but not in  PBS-control treated samples. The test statistic for the interaction test between PMA treatment and live cell concentration was evaluated. The triplicated experiments were performed on four species, but species was non-significant in a three-way interaction. Species was retained as fixed factor, and the denominator degrees of freedom were estimated according to Satterthwaite (ANOVA) or Kenward-Roger approximations (slope estimates) for final inference. 

The threeway interaction term with species identity is not significant (F(`r round(anova_lmm_test$NumDF[7], digits = 0)`,`r round(anova_lmm_test$DenDF[7], digits = 0)`) = `r round(anova_lmm_test$F.statistic[7], digits = 2)`, *p* = `r round(anova_lmm_test$p.value[7], digits = 2)`). The model is updated with species retained only as fixed effect, and assessment of the interaction between PMA treatment and life cell fraction is evaluated as the mean across species.

```{r ExpI_LMM random effect assessment, warning=F, message=F}

# LMM goodness of fit evaluation (marginal = variance explained by fixed effects; conditional = by fixed and random effects combined)
variance_components = MuMIn::r.squaredGLMM(model_lmer) %>% data.frame
variance_components

# assess contribution of random variables to total variance using likelihood ratio test (LRT)
ranova_lmm = ranova(model_lmer)
confint_terms = confint(model_lmer, oldNames =F)

# format LMM anova
lrt_lmm =
  ranova_lmm %>% 
  broom::tidy() %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     TRUE ~ ""))

# view LRT
lrt_lmm
summ_lmm = summary(model_lmer)

# % variance explained by random intercept
re_var = 
  as_tibble(summ_lmm$varcor) %>%  
  summarise(re_perc = 100*vcov[1]/sum(vcov)) %>% 
  pull(re_perc) 

# save LRT
lrt_lmm %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv("output_data/ELLY_DPOL_ExpI_general_LMM_LRT_results.csv")


```

The starting concentrations of overnight cell culture suspensions varied between experimental dates, and a random intercept for date (factor) was included in the model, accounting for `r round(re_var, digits = 0)`% of variance (likelihood ratio test; `r paste0("\u03C7","\u00B2")` = `r round(lrt_lmm$LRT[2], digits =1)`, df = `r round(lrt_lmm$df[2], digits =0)`, *p* = `r paste0("10<sup>", sprintf("%.0f", log10(lrt_lmm$p.value[2])), "</sup>")`).

```{r ExpI_model assumptions I, warning=F, message=F}

## test model assumptions for model_lmer
model.metrics <- augment(model_lmer)
# Assess normality of residuals using Kolmogorov-Smirnov test (model df = 101 > 50)
ks_test <- ks.test(model.metrics$.resid, 
                   y = "pnorm", 
                   mean = mean(model.metrics$.resid), 
                   sd = sd(model.metrics$.resid)
                   )

# reshape KS results
ks_test_df = 
ks_test %>% 
 broom::tidy() %>% 
  dplyr::select(method, alternative, statistic, p.value) %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     p.value > 0.05 ~ "n.s.",
                     TRUE ~ ""))

# view KS results
ks_test_df

# save KS results
ks_test_df %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv("output_data/ELLY_DPOL_ExpI_general_LMM_Kolmogorov_Smirnov_results.csv")

```
Kolmogorov-Smirnov test of residuals shows no deviation from normality of LMM residuals (statistic = `r round(ks_test_df$statistic[1], digits = 2)`, two-sided *p* = `r round(ks_test_df$p.value[1], digits = 2)`).

```{r ExpI_model assumptions II, warning=F, message=F}


model.metrics <- augment(model_lmer)
# Assess normality of residuals using Shapiro-Wilk test 
sw_test = shapiro_test(model.metrics$.resid)

# reshape SW results
sw_test_df = 
  sw_test %>% 
  mutate(method = "Shapiro-Wilk normality test") %>% 
  dplyr::select(method, "W.statistic" = statistic, p.value) %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     p.value > 0.05 ~ "n.s.",
                     TRUE ~ ""))


# view SW results
sw_test_df

# save SW results
sw_test_df %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv("output_data/ELLY_DPOL_ExpI_general_LMM_Shapiro_Wilk_results.csv")

```
Shapiro-Wilk test shows no deviation from normality of LMM residuals (statistic = `r round(sw_test_df$W.statistic[1], digits = 2)`, *p* = `r round(sw_test_df$p.value[1], digits = 2)`).

```{r ExpI_model assumptions III, warning=F, message=F}

## Assess homogeneity of variance
## plot fitted vs. residuals
# ggplot(model.metrics,
#        aes(x=.fitted, y=.resid)) +
#   geom_point() +
#   geom_smooth(se =F, method = "lm") +
#   facet_wrap(~pma)

# Levene's test for homogeneity of variances
levene = leveneTest(residuals(model_lmer) ~ interaction(data_lm_expI$pma, data_lm_expI$conc))

# reshape leveneTest results
levene_df = 
levene %>% 
  broom::tidy() %>% 
  mutate(method = "Levene's test") %>% 
  dplyr::select(method, "F.statistic" = statistic, df, df.residual, p.value) %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     p.value > 0.05 ~ "n.s.",
                     TRUE ~ ""))

# view leveneTest result
levene_df

# save leveneTest result
levene_df %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv("output_data/ELLY_DPOL_ExpI_general_LMM_Levene_test_results.csv")

```
Levene's test for homogeneity of variance shows no deviation from the expected homoscedasticity (F(`r round(levene_df$df[1], digits = 0)`,`r round(levene_df$df.residual[1], digits = 0)`) = `r round(levene_df$F.statistic[1], digits = 2)`, *p* = `r round(levene_df$p.value[1], digits = 2)`).

A high goodness of fit was achieved for the linear mixed model with a conditional `r paste0("R","\u00B2")` of `r round(variance_components[2]*100, digits=0)`%.

```{r ExpI_model inference, warning=F, message=F}

# calculate overall trends for PMA and none 
overall_trend = emtrends(model_lmer, pairwise ~ pma, var = "log_conc")

# reformat PMA trend
overall_trend_df = 
  overall_trend$emtrends %>% 
  broom::tidy() %>% 
  rename(F.statistic = "statistic")

# save slope stats per species
overall_trend_df %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv(file = "output_data/ELLY_DPOL_ExpI_anova_results_overall_per_treatment_slopes.csv", col_names = T)

# view pma treatment trend contrast
overall_contrast_df = 
  overall_trend$contrasts %>% 
  broom::tidy() %>% 
  rename(F.statistic = "statistic")

 overall_contrast_df

# save slope stats per species
overall_contrast_df %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv(file = "output_data/ELLY_DPOL_ExpI_anova_results_treatment_contrast_slopes.csv", col_names = T)


## Only interpret results below when threeway interaction with species is significant; see anova_lmm
      # # extract slopes for pma treatment groups per species
      # species_trends = emtrends(model_lmer, pairwise ~ pma | species, var = "log_conc")
      # 
      # # show slope stats and t tests for pma treatment contrast
      # species_trends
      # 
      # # average slopes PMA
      # # save slope stats per species
      # species_trends$emtrends %>% 
      #   broom::tidy() %>% 
      #   group_by(pma) %>% 
      #   summarise(slope = mean(log_conc.trend),
      #             se = mean(std.error))
      #             
      # 
      # # save slope stats per species
      # species_trends$emtrends %>% 
      #   broom::tidy() %>% 
      #   mutate(p.value.format = format(p.value, scientific=T)) %>% 
      #   mutate_if(is.numeric, ~ round(., 3)) %>% 
      #   write_excel_csv(file = "output_data/ELLY_DPOL_ExpI_anova_species_specific_per_treatment_slopes.csv", col_names = T)
      # 
      # 
      # # show stats
      # species_trends$contrasts
      # 
      # # save PMA vs PBS control stats within species
      # species_trends$contrasts %>% 
      #   broom::tidy() %>% 
      #   mutate(p.value.format = format(p.value, scientific=T)) %>% 
      #   mutate_if(is.numeric, ~ round(., 3)) %>% 
      #   write_excel_csv(file = "output_data/ELLY_DPOL_ExpI_anova_results_difference_in_slopes_within_species.csv", col_names = T)
      # 
      # 
      # # run ancova model to compare PMA slopes among species
      # pma_trends = emtrends(model_test_lmer,  pairwise ~ species | pma, var = "conc")
      # 
      # pma_trends$contrasts %>% 
      #   broom::tidy() %>% 
      #   mutate(p.value.format = format(p.value, scientific=T)) %>% 
      #   mutate_if(is.numeric, ~ round(., 3)) %>% 
      #   filter(pma == "PMA") 
      #   
      # pma_trends$contrasts %>% 
      #   broom::tidy() %>% 
      #   mutate(p.value.format = format(p.value, scientific=T)) %>% 
      #   mutate_if(is.numeric, ~ round(., 3)) %>% 
      #   write_excel_csv(file = "output_data/ELLY_DPOL_ExpI_anova_results_difference_in_PMA_slopes_among_species.csv", col_names = T)

```
The interaction between PMA treatment and  proportion of live cells was significant (interaction: F(`r round(anova_lmm$NumDF[4], digits = 0)`,`r round(anova_lmm$DenDF[4], digits = 0)`) = `r round(anova_lmm$F.statistic[4], digits = 2)`, *p* = `r paste0("10<sup>", sprintf("%.0f", log10(anova_lmm$p.value[4])), "</sup>")`, with a slope (SE) for PMA pre-treatment of `r round(as_tibble(overall_trend$emtrends)$log_conc.trend[2], digits = 2)` (`r round(as_tibble(overall_trend$emtrends)$SE[2], digits = 2)`) that was significantly different from control treatment with a slope of `r round(as_tibble(overall_trend$emtrends)$log_conc.trend[1], digits = 2)` (`r round(as_tibble(overall_trend$emtrends)$SE[1], digits = 2)`) (slopes control-PMA = `r round(as_tibble(overall_trend$contrasts)$estimate, digits = 2)`; t = `r round(as_tibble(overall_trend$contrasts)$t.ratio, digits = 1)`, df = `r round(as_tibble(overall_trend$contrasts)$df, digits = 0)`, *p* = `r paste0("10<sup>", sprintf("%.0f", log10(as_tibble(overall_trend$contrasts)$p.value)), "</sup>")`).  

```{r ExpI_test consistency of marginal effect tests, echo=F, eval=F, include=F}

## sanity check
# only when threeway interactions with species is significant in model_test_lmer
# verify estimates with different methods
species_trends$contrasts

# library phia
library(phia)
phia::interactionMeans(model_test_lmer,slope = "log_conc")
phia::testInteractions(model_test_lmer,pairwise = "pma", fixed = "species", slope = "log_conc")

# library marginaleffects
library(marginaleffects)
slopes(model_test_lmer, variables = "log_conc", by = c("pma","species")) %>% as_tibble() %>% dplyr::select(-term, -contrast) %>% arrange(species, pma)

```

```{r ExpI_plot slopes on data, eval=F, echo=F, include=F, warning=F, message=F}

plot_4sp = 
 data_clean %>%
   group_by(species) %>% 
   arrange(exp_date) %>% 
   mutate(Replicate = LETTERS[match(exp_date, unique(exp_date))]) %>% 
      
  ggplot(aes(x=conc, y=sq_calc_mean, color=pma, label = pma)) + # y=pred for fixed effect predictions
  geom_point(aes(shape=Replicate)) +
  #geom_line(data = line, aes(x=x, y=y), color = "grey50", inherit.aes = F) +
  scale_color_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  scale_fill_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x)),
                limits = c(10^4, NA)) +
  scale_x_continuous(trans = "log1p",
                breaks = c(0.1, 1, 10, 50, 100),
                labels = c(0.1, 1, 10, 50, 100),
                limits = c(0, 100)
               ) +
  scale_shape_manual(values = c(0, 1,2, 16,23,20,5,7,17,18, 19, 21, 25,12, 13)) +
  geom_smooth(aes(linetype = pma, fill = pma),method = "lm", se = T, show.legend = F) + 
  labs(title = "Effect of PMA treatment on qPCR amplification",
       # subtitle = "*PMA treatment improves accuracy of PCR-based methods<br>when a liquid sample contains relatively much (isopropanol) killed material*",
       y = "log 16S rRNA gene copies mL^(-1)",
       x = "Proportion of live cells (%)") +
  theme_light() +
  theme(axis.title.y = element_markdown(),
        plot.subtitle = element_markdown(colour = "#00BFC4"),
        strip.text = element_markdown(margin = margin(t = 10)), 
        strip.background = element_rect(color = NA)
  ) +
  coord_cartesian(clip = "off") + 
  facet_wrap(~species)



plot_4sp

```

Calculate the expected target gene copy concentrations when PMA efficacy is 100%. Species-specific and PMA treatment-specific expectations were based on empirical measurements of 100% live cells. For untreated PBS control samples, the expected values are uniform across the range of life cell concentrations since conventional qPCR is expected not to discriminate target DNA from life, damaged and dead cells. For PMA treated samples, the expected values are calcuated as reductions proportional to the life cell concentration.

```{r ExpI_theoretical expectation reduction with PMA, eval=T, echo=T, include=T, warning=F, message=F}

# y values per exp_id expected when PMA is 100% efficacious
params_4sp = 
  data_clean %>% 
  filter(!grepl("2022_07_12|2022_08_10", exp_date)) %>% 
  group_by(pma, conc, species) %>% 
  filter(conc == 100,
         pma == "PMA") %>% 
  ungroup() %>% 
  group_by(species) %>% 
  summarise(mean_life = mean(sq_calc_mean))

# y values for PBS control when based on average value with 100% live cells
params_noPMA_4sp = 
  data_clean %>% 
  group_by(pma, conc, species) %>% 
  filter(conc == 100,
         pma == "none") %>% 
  ungroup() %>% 
  group_by(species) %>% 
  summarise(mean_life = mean(sq_calc_mean))

# Define the x value range for fitting theoretical curve
x_values <- c(seq(100, 2, -1), seq(1, 0.2, -0.1), seq(0.19, 0.01, -0.01))

# Create a named list of empirical starting y values
initial_y_values <- list(
  `*Enterococcus faecalis*` = params_4sp$mean_life[1],
  `*Escherichia coli*` =  params_4sp$mean_life[2],  
  `*Pseudomonas citronellolis*` =  params_4sp$mean_life[3],   
  `*Stenotrophomonas rhizophila*` =  params_4sp$mean_life[4]   
)

# Function to generate data frame with x and y values based on log-linear decay
generate_data <- function(initial_y, x_values) {
  # Define the power-law prediction function
  log_y_values <- log10(initial_y) - (log10(x_values) - log10(max(x_values))) * log10(0.1) 
  predicted_y = 10^log_y_values
  result = tibble(x= x_values, predicted_y = predicted_y)
  return(result)
}


# Apply the function to each named element in the list
result_list_4sp <- lapply(initial_y_values, generate_data, x_values = x_values)

# Combine the results into a single data frame
result_df_4sp <- do.call(bind_rows, result_list_4sp)
#result_df_4sp$exp_id = row.names(result_df_4sp)
result_df_4sp = 
  as_tibble(result_df_4sp) %>% 
  mutate(species = factor(rep(names(result_list_4sp), 
                             each = length(x_values))))
                    

```

Species-specific concentrations are presented on a log10-transformed y-axis as a function of life cell fraction in percentage. The theoretical 100% efficacy (PMA) and 0% efficacy (PBS control) are plotted as solid lines, with ordinary least squares linear model fits plotted (Note that the statistical analysis for PMA efficacy inference corrects for random variance attributable to experimental dates (21% of total variance). 

```{r ExpI_plot qPCR observed and predicted data, warning=F, message=F}

# create plot with predicted 100% PMA efficacy curves with data added
plot_DPOL_ELLY_4sp = 
data_clean %>% 
  mutate(conc = ifelse(conc == 0, 0.01, conc)) %>% 
  ggplot(aes(x= conc, y = sq_calc_mean, color=pma)) +
  geom_line(data = result_df_4sp %>% 
              mutate(sq_calc_mean = predicted_y) %>% 
              filter(x > 0.01),
            aes(x = x, y = sq_calc_mean), 
            color = pal[6], linewidth = 0.5,  inherit.aes = F) +
  geom_segment(data = params_noPMA_4sp %>% mutate(sq_calc_mean = mean_life), 
            aes(y = sq_calc_mean, yend = sq_calc_mean, x = 0.01, xend = 100), 
            color = pal[7], linewidth = 0.5) +
  geom_smooth(aes(fill = pma), linetype = 2, size = 0.75, method = "lm", se = T, show.legend = F, alpha = 0.5) +
  geom_point(shape = 21, fill="white", show.legend = F) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x)),
                limits = c(10^2, 10^10)) +
  scale_x_log10(breaks = c(0.01, 0.1, 1, 10, 50, 100),
                labels = c(0, 0.1, 1, 10, 50, 100)) +
  # scale_x_continuous(trans = "log10",
  #                    breaks = c(0.01, 0.1, 1, 10, 50, 100),
  #                    labels = c(0, 0.1, 1, 10, 50, 100),
  #                    limits = c(0, 100)) +
  scale_color_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  scale_fill_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  facet_wrap(~species) +
  labs(x = "Proportion of live cells (%)",
       y = bquote("log"[10] * " 16S rRNA gene copies mL"^-1 *"")) +
  theme_light() +
  theme(strip.text = element_markdown()) +
  geom_text(data = data_clean %>%
              ungroup() %>% filter(species == "*Enterococcus faecalis*") %>% 
              slice_head(n=1),
            aes(x = 90, y=10^6), label = "PMA-qPCR", color = pal[6],
            hjust =1,
            size = 3) +
  geom_text(data = data_clean %>%
              ungroup() %>% filter(species == "*Enterococcus faecalis*") %>% 
              slice_head(n=1),
            aes(x = 0.01, y=10^8.8), label = "qPCR", color = pal[7],
            hjust =0,
            size = 3) +
  geom_text(data = data_clean %>% 
                   filter(species == "*Enterococcus faecalis*") %>% 
                   slice_head(n=1),
            aes(x = 9, y=700), 
            label = "100% PMA efficacy", 
            color = "grey50", 
            hjust = 0,
            size = 2.5) +
  geom_segment(data = data_clean,
               aes(y = 700, yend = 700,
                   x = ifelse(species == "*Enterococcus faecalis*", 4, 0),
                   xend = ifelse(species == "*Enterococcus faecalis*", 8, 0)),
               color = "grey50",
               linewidth = 0.5,
               inherit.aes = F) +
  geom_text(data = data_clean %>% 
                   ungroup() %>% 
                   filter(species == "*Enterococcus faecalis*") %>% 
                   slice_head(n=1),
            aes(x = 9, y=250), 
            label = "LM fit (±SE)", 
            color = "grey50", 
            hjust =0,
            size = 2.5) +
  geom_segment(data = data_clean, 
               aes(y = 250, yend = 250, 
                   x = ifelse(species == "*Enterococcus faecalis*", 4, 0), 
                   xend = ifelse(species == "*Enterococcus faecalis*", 8, 0)), 
               color = "grey50",
               linetype = 2, 
               linewidth = 0.5, 
               inherit.aes = F)
  

plot_DPOL_ELLY_4sp

ggsave(plot = plot_DPOL_ELLY_4sp, filename = "figures/DPOL_ELLY__ExpI_PMA_effect_4species_lmfit_theoretical_line.pdf", width = 7, height = 6)

```

Fig. Species-specific *in vitro* test of PMA efficacy of type strains of *E.coli*, *E.faecalis*, and isolated *P.citronellolis* and *S.rhizophila* as function of life cell percentage in the mixture.



### Experiment II: PMA efficacy at different starting concentrations of *E. coli*. High concentration is data from Exp. I and compared to medium and low starting cell concentrations.

Theoretical expectation at 100% efficacy is calculated and modeled.

```{r theoretical expectation reduction with PMA starting concentrations, warning=F, message=F}

# y values per exp_id expected when PMA is 100% efficacious
params = 
  data_clean4 %>% 
  mutate(pma = if_else(pma == "no_PMA", "no PMA", pma),
         exp_id = factor(exp_id, 
                         levels = c("E.coli_low", "E.coli_medium", "E.coli_high"))) %>% 
  group_by(pma, conc, exp_id) %>% 
  mutate(sq_calc_mean = 10^log_sq_calc_mean) %>% 
  filter(conc == 100,
         pma == "PMA",
         ) %>% 
  unique %>% 
  ungroup() %>% 
  group_by(exp_id) %>% 
  summarise(mean_life = mean(sq_calc_mean))

# y values for PBS control when based on average value with 100% live cells
params_noPMA = 
  data_clean4 %>% 
  mutate(pma = if_else(pma == "no_PMA", "no PMA", pma),
         exp_id = factor(exp_id, 
                         levels = c("E.coli_low", "E.coli_medium", "E.coli_high"))) %>% 
  group_by(pma, conc, exp_id) %>% 
  mutate(sq_calc_mean = 10^log_sq_calc_mean) %>% 
  filter(conc == 100,
         pma == "none") %>% 
  unique %>% 
  ungroup() %>% 
  group_by(exp_id) %>% 
  summarise(predicted_y = mean(sq_calc_mean))


# Define the x value range for fitting theoretical curve
x_values <- c(seq(100, 2, -1), seq(1, 0.2, -0.1), seq(0.19, 0, -0.01))

# Create a named list of empirical starting y values
initial_y_values <- list(
  "E.coli_low" = params$mean_life[1],
  "E.coli_medium" =  params$mean_life[2],  # Adjust this value as needed
  "E.coli_high" =  params$mean_life[3]   # Adjust this value as needed
)

# Function to generate data frame with x and y values based on logarithmic decay model
generate_data <- function(initial_y, x_values) {
  # Define the power-law prediction function
  log_y_values <- log10(initial_y) - (log10(x_values) - log10(max(x_values))) * log10(0.1) 
  predicted_y = 10^log_y_values
  result = tibble(x= x_values, predicted_y = predicted_y)
  return(result)
}
# Apply the function to each named element in the list
result_list <- lapply(initial_y_values, generate_data, x_values = x_values)

# Combine the results into a single data frame
result_df <- do.call(bind_rows, result_list)
result_df$exp_id = row.names(result_df)
result_df = 
  as_tibble(result_df) %>% 
  mutate(exp_id = factor(rep(names(result_list), 
                             each = length(x_values))))
                    

```


```{r plot observed and predicted data starting concentrations, warning=F, message=F}

# reshape to plot data starting concentrations
data_clean5 = 
data_clean4 %>% 
    group_by(pma, conc, exp_id, date) %>% 
    mutate(sq_calc_mean = 10^log_sq_calc_mean,
           pma = if_else(pma == "no_PMA", "no PMA", pma),
           exp_id = factor(exp_id, 
                           levels = c("E.coli_low", "E.coli_medium", "E.coli_high")),
           conc = ifelse(conc == 0, 0.1, conc)) %>% 
    unique() 


# plot data of different Total bacterial cell concentrations + lm fit + theoretical PMA effect 
plot_total_cells_NMID_DPOL = 
data_clean5 %>%
 ggplot(aes(x= conc, y = sq_calc_mean, color=pma)) +
  geom_line(data = result_df %>% 
              mutate(sq_calc_mean = predicted_y) %>% 
              filter(x > 0.01),
            aes(x = x, y = sq_calc_mean), 
            color = pal[6], linewidth = 0.5,  inherit.aes = F) +
  geom_segment(data = params_noPMA %>% mutate(sq_calc_mean = predicted_y), 
            aes(y = sq_calc_mean, yend = sq_calc_mean, x = 0.01, xend = 100), 
            color = pal[7], linewidth = 0.5) +
  geom_smooth(aes(fill = pma), linetype = 2, size = 0.75, 
              method = "lm", se = T, 
              #formula = y ~ poly(x, 2),
              show.legend = F, alpha = 0.5) +
  geom_point(shape = 21, fill="white", show.legend = F) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x)),
                limits = c(10^1, 10^10)) +
  scale_x_log10(breaks = c(0.01, 0.1, 1, 10, 50, 100),
                labels = c(0, 0.1, 1, 10, 50, 100)) +
  # scale_x_continuous(trans = "log10",
  #                    breaks = c(0.01, 0.1, 1, 10, 50, 100),
  #                    labels = c(0, 0.1, 1, 10, 50, 100),
  #                    limits = c(0, 100)) +
  scale_color_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  scale_fill_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
 facet_wrap(~exp_id, 
             labeller = labeller(exp_id = c(E.coli_low = "Low cell concentration", 
                                            E.coli_medium = "Medium cell concentration", 
                                            E.coli_high = "High cell concentration"))) +
  labs(x = "Proportion of live cells (%)",
       y = bquote("log"[10] * " 16S rRNA gene copies mL"^-1 *"")) +
  theme_light() +
  geom_text(data = data_clean5 %>% 
                   ungroup() %>% 
                   filter(exp_id == "E.coli_high") %>% 
                   slice_head(n=1),
            aes(x = 90, y=10^6.7), 
            label = "PMA-qPCR", 
            color = pal[6], 
            hjust =1,
            size = 3) +
  geom_text(data = data_clean5 %>% 
                   ungroup() %>% 
                   filter(exp_id == "E.coli_high") %>% 
                   slice_head(n=1),
            aes(x = 0.01, y=10^9.5), 
            label = "qPCR", 
            color = pal[7], 
            hjust =0,
            size = 3) +
  geom_text(data = data_clean5 %>% 
                   ungroup() %>% 
                   filter(exp_id == "E.coli_high") %>% 
                   slice_head(n=1),
            aes(x = 2, y=500), 
            label = "100% PMA efficacy", 
            color = "grey50", 
            hjust = 0,
            size = 2.5) +
  geom_segment(data = data_clean5,
               aes(y = 500, yend = 500,
                   x = ifelse(exp_id == "E.coli_high", 0.5, 0),
                   xend = ifelse(exp_id == "E.coli_high", 1.5, 0)),
               color = "grey50",
               linewidth = 0.5,
               inherit.aes = F) +
  geom_text(data = data_clean5 %>% 
                   ungroup() %>% 
                   filter(exp_id == "E.coli_high") %>% 
                   slice_head(n=1),
            aes(x = 2, y=200), 
            label = "LM fit (±SE)", 
            color = "grey50", 
            hjust =0,
            size = 2.5) +
  geom_segment(data = data_clean5, 
               aes(y = 200, yend = 200, 
                   x = ifelse(exp_id == "E.coli_high", 0.5, 0), 
                   xend = ifelse(exp_id == "E.coli_high", 1.5, 0)), 
               color = "grey50",
               linetype = 2, 
               linewidth = 0.5, 
               inherit.aes = F)
  

# view plot
plot_total_cells_NMID_DPOL

# save plot
ggsave(plot = plot_total_cells_NMID_DPOL, filename = "figures/NMID_ExpII_total_cell_conc_effect_on_PMA_NMID_DPOL_updated.pdf", width = 8, height = 4)


```
Fig. Effect of starting cell concentration on PMA efficacy.

```{r ExpII_model_total_cell_effect, warning=F, message=F}

# ungroup data if needed
data_clean6 = data_clean5 %>% unique %>% ungroup

# create linear model
lmer_total_cell_effect =
  with(data_clean6, 
       lmer(log_sq_calc_mean ~ conc*pma*exp_id + (1|date), REML = T))

# LMM goodness of fit evaluation (marginal = variance explained by fixed effects; conditional = by fixed and random effects combined)
variance_components_expII = MuMIn::r.squaredGLMM(lmer_total_cell_effect) %>% data.frame
variance_components_expII

# assess contribution of random variables to total variance using likelihood ratio test (LRT)
ranova_lmm_expII = ranova(lmer_total_cell_effect)
confint_terms_expII = confint(lmer_total_cell_effect, oldNames =F)

# format LMM anova
lrt_lmm_expII =
  ranova_lmm_expII %>% 
  broom::tidy() %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     TRUE ~ ""))

# view LRT
lrt_lmm_expII
summ_lmm_expII = summary(lmer_total_cell_effect)

# % variance explained by random intercept
re_var_expII = 
  as_tibble(summ_lmm_expII$varcor) %>%  
  summarise(re_perc = 100*vcov[1]/sum(vcov)) %>% 
  pull(re_perc) 

# save LRT
lrt_lmm_expII %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv("output_data/NMID_ExpII_starting_conc_LMM_LRT_results.csv")

```

```{r ExpII_model assumptions I, warning=F, message=F}


model.metrics_expII <- augment(lmer_total_cell_effect)
# Assess normality of residuals using Shapiro-Wilk test 
sw_test_expII = shapiro_test(model.metrics_expII$.resid)

# reshape SW results
sw_test_df_expII = 
  sw_test_expII %>% 
  mutate(method = "Shapiro-Wilk normality test") %>% 
  dplyr::select(method, "W.statistic" = statistic, p.value) %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     p.value > 0.05 ~ "n.s.",
                     TRUE ~ ""))


# view SW results
sw_test_df_expII 

# significant but nicely clock shaped, would not harm inference.
hist(model.metrics_expII$.resid)

# save SW results
sw_test_df_expII %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>%
  write_excel_csv("output_data/NMID_ExpII_general_LMM_Shapiro_Wilk_results.csv")

```


```{r ExpII_model assumptions II,  warning=F, message=F}

## Assess homogeneity of variance
## plot fitted vs. residuals
# ggplot(model.metrics,
#        aes(x=.fitted, y=.resid)) +
#   geom_point() +
#   geom_smooth(se =F, method = "lm") +
#   facet_wrap(~pma)

# Levene's test for homogeneity of variances
levene_expII = leveneTest(residuals(lmer_total_cell_effect) ~ interaction(data_clean6$pma, data_clean6$conc))

# reshape leveneTest results
levene_df_expII = 
levene_expII %>% 
  broom::tidy() %>% 
  mutate(method = "Levene's test") %>% 
  dplyr::select(method, "F.statistic" = statistic, df, df.residual, p.value) %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     p.value > 0.05 ~ "n.s.",
                     TRUE ~ ""))

# view leveneTest result
levene_expII

# significant but not a plume shape and trend horizontal, would not harm inference.
plot(model.metrics_expII$.fitted, model.metrics_expII$.resid)

# save leveneTest result
levene_df_expII %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>%
  write_excel_csv("output_data/NMID_ExpII_general_LMM_Levene_test_results.csv")

```

```{r ExpII_model inference, warning=F, message=F}

# overall anova results
anova(lmer_total_cell_effect)
#summary(lmer_total_cell_effect)

# save anova results
anova(lmer_total_cell_effect) %>% 
 broom::tidy() %>% 
  as_tibble() %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>%
  write_excel_csv(file = "output_data/NMID_ExpII_total_cell_concentration_anova_results.csv", col_names = T)

# overall emmeans H0 = 0
emm_expII = emmeans(lmer_total_cell_effect, ~ conc*pma | exp_id, infer=T)

# Compare slopes within groups (e.g., comparing the slope in group A to group B)
emtrend_within_expII = emtrends(lmer_total_cell_effect, pairwise ~ pma | exp_id, var = "conc", infer =TRUE)

# view result
emtrend_within_expII

# save output table ancova
emtrend_within_expII$contrasts %>% 
  broom::tidy() %>% 
  as_tibble() %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>%
  write_excel_csv(file = "output_data/NMID_SI_ExpII_Total_cell_concentration_anova_slopes_within_group.csv", col_names = T)

## compare slopes of PMA between exp_ids

# Compare slopes within groups (e.g., comparing the slope in group A to group B)
emtrend_between_expII = emtrends(lmer_total_cell_effect, pairwise ~ exp_id | pma, var = "conc", infer =TRUE, adjust = "Holm")

# view result
emtrend_between_expII

# save output table ancova
emtrend_between_expII$contrasts %>% 
  broom::tidy()  %>% 
  mutate(adj.p.value.format = format(adj.p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv(file = "output_data/NMID_SI_ExpII_Total_cell_concentration_anova_PMA_slopes_between_expIDs.csv", col_names = T)


```




### Experiment III: PMA efficacy on ozonated wastewater treatment plant effluent spiked with *E. coli* and *P. aeruginosa*

#### Bacterial survival of spiked strains based on culturing 

Survival differences of  *E. coli* and *P. aeruginosa* after ozone treatment of spiked wastewater treatment plant effluent was assessed using ordinary least squares linear regression. The response was log10 transformed colony forming units per 100 mL filtered water, and a two-way interaction was tested between species and the sample_type (spiked effluent vs ozone)

```{r ExpIIII_statistics culturing - model, warning=F, message=F}

# transform culturing based concentrations to log10 scale
culture_data2 = 
  culture_data %>% 
  mutate(log_mean_CFU_100mL = log10(mean_CFU_100mL))

# formulate linear regression model culturing
lm_culture = lm(log_mean_CFU_100mL ~ sample_type*species, culture_data2)

# reshape anova culturing
anova_culture = 
  anova(lm_culture) %>% 
  broom::tidy() %>% 
  mutate(across(c(sumsq:statistic)
                , ~round(.x, digits = 3))) %>% 
  rename("F.statistic" = statistic) %>% 
  mutate(
    p.value.format = format(p.value, scientific=T),
    sig = case_when(p.value < 0.0001 ~ "****",
                    p.value < 0.001 ~ "***",
                    p.value < 0.01 ~ "**",
                    p.value < 0.05 ~ "*",
                    TRUE ~ ""))

# view anova culturing
anova_culture 

# save anova culturing
anova_culture %>% 
  write.xlsx(file = "output_data/ELLY_DPOL_ExpIII_anova_output_culturing_log_CFU-100mL_between_methods.xlsx",
             overwrite = T)

```

```{r ExpIIII_statistics culturing - contrasts, warning=F, message=F}


# estimate marginal means and 95% CI
emmeans_culture = 
  emmeans(lm_culture, ~sample_type*species) 


## means estimated per sample_type and species
# view culture marginal means
emmeans_culture_df = 
emmeans_culture %>% 
  broom::tidy() %>% 
  rename("t.statistic" = statistic) %>% 
  mutate(
    sig = case_when(p.value < 0.0001 ~ "****",
                    p.value < 0.001 ~ "***",
                    p.value < 0.01 ~ "**",
                    p.value < 0.05 ~ "*",
                    TRUE ~ ""))

emmeans_culture_df

# save culture marginal means
emmeans_culture_df %>% 
  write.xlsx(file = "output_data/ELLY_DPOL_ExpIII_marginal_means_culturing_log_CFU-100mL_between_methods.xlsx",
             overwrite = T)
  

## pairwise differences to assess the reduction by ozonationa and how that differs between the spiked species

# evaluate sample_type differences within species as Tukey post-hoc (Tukey P.adj)
pairs_culture = 
  pairs(emmeans_culture, by = "species") %>%  
  broom::tidy() %>%
  rename("t.statistic" = statistic) %>% 
  mutate(
    
    sig = case_when(adj.p.value < 0.0001 ~ "****",
                    adj.p.value < 0.001 ~ "***",
                    adj.p.value < 0.01 ~ "**",
                    adj.p.value < 0.05 ~ "*",
                    TRUE ~ ""))

pairs_culture

# save
pairs_culture %>% 
  write.xlsx(file = "output_data/ELLY_DPOL_ExpIII_Tukey_posthoc_between_sampletypes_culturing_log_CFU-100mL_between_methods.xlsx",
             overwrite = T)


```


```{r ExpIII_model assumptions culturing I, warning=F, message=F}

model.metrics_culturing <- augment(lm_culture) 

# Assess normality of residuals using Shapiro-Wilk test 
sw_test_culturing = shapiro_test(model.metrics_culturing$.resid)

# reshape SW results
sw_test_culturing_df = 
  sw_test_culturing %>% 
  mutate(method = "Shapiro-Wilk normality test") %>% 
  dplyr::select(method, "W.statistic" = statistic, p.value) %>% 
  mutate(
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     p.value > 0.05 ~ "n.s.",
                     TRUE ~ ""))


# view SW results
sw_test_culturing_df

# save SW results
sw_test_df %>% 
  mutate(p.value.format = format(p.value, scientific=T)) %>% 
  mutate_if(is.numeric, ~ round(., 3)) %>% 
  write_excel_csv("output_data/ELLY_DPOL_ExpIII_general_LMM_Shapiro_Wilk_results.csv")

```

```{r ExpIII_plot raw culturing data, warning=F, message=F}

# table mean counts per sample type
culture_data %>% 
  mutate(sample_type = factor(sample_type,
                              levels = c("effluent", "spike_culture", "effluent_spike", "ozone"),
                              labels = c("effluent", "culture", "effluent spiked", "ozone")),
         species = factor(species,
                          levels = c("E.coli", "P.aeruginosa"),
                          labels = c("*E. coli*", "*P. aeruginosa*"))) %>% 
  group_by(species,sample_type) %>% 
  summarise(mean_cfu_100mL = mean(mean_CFU_100mL),
            sd_cfu_100mL = sd(mean_CFU_100mL),
            log_mean_cfu_100mL = log10(mean_cfu_100mL)) %>% 
  write_csv("output_data/ELLY_DPOL_ExpIII_ExpIII_table_log10_means.csv")

# plot raw culturing data
plot_culturing_df <- 
culture_data %>% 
  mutate(sample_type = factor(sample_type,
                              levels = c("effluent", "spike_culture", "effluent_spike", "ozone"),
                              labels = c("effluent", "culture", "effluent spiked", "ozone")),
         species = factor(species,
                          levels = c("E.coli", "P.aeruginosa"),
                          labels = c("*E. coli*", "*P. aeruginosa*"))) %>% 
ggplot( aes(x = sample_type, 
            y = log10(mean_CFU_100mL), 
            color = species)) +
  geom_boxplot(aes(fill = species),
               outlier.shape = NA, alpha=0.5) + 
  geom_point(aes(color = factor(species), shape = factor(exp_date)),
             position = position_jitterdodge(dodge.width = 0.75, jitter.width = 0),
             alpha = 1, size = 2) +
  ylab(expression(log[10] ~ 'mean' ~ '[CFU' ~ '100 mL'^-1 ~ ']')) +
  xlab("Sample") +
  theme_light() +
  theme(axis.text.x = element_text(vjust = 0.5, hjust=0.5), 
        legend.position="right",
        panel.grid.major.x = element_blank(),
        legend.text = element_markdown(),
        axis.title.y = element_text(size = 12),
        axis.title.x = element_text(size = 12),
        panel.grid.minor = element_blank()) +
  scale_y_continuous(breaks = seq(0, 13, by = 1)) +
  scale_color_manual(name = "Species", values = c("#F3B61F", "#2AB7CA")) +
  scale_fill_manual(name = "Species", values = c("#F3B61F", "#2AB7CA")) +
  scale_shape_discrete(name = "Date")

# print plot
plot_culturing_df

# save plot
ggsave("figures/ELLY_DPOL_ExpIII_plot_culturing_df.pdf", plot_culturing_df, width = 6, height = 4)

```

#### qPCR based analysis of bacterial survival of spiked strains

##### qPCR standard curve assessment and QC

First standard curves were analyzed as quality control step.

```{r quality control standard_curve data qPCR, warning=F, message=F}

# Observations:
# - One of the standard (Std-07) seems to be off
# - All data points are in between the standards
# - 1 effluent data point is under the highers CT standard in PA_marta.
# Blanks:
# - 14 negative controls were negative
#   - Content NTC-01: 
#     - for PA_marta and Ybbw primers they were all NA Cq, 
#     - for 16S_gene the Cq ranged from 29.5-30.1, but no starting quantity.
# - Blanks CT are within the standard curve, but also not able to measure in the specific primer sets.
# - Blanks on 16S primer with no score is the lowest mean CT 29.83786, and singe data point 29.79716 CT.
#   - cutoff CT < 29.79716
# 
# Limit of Detection (LoD):
# - common formula: LOD = mean NTC Ct + (3 * standard deviation NTC)
# 
# ## quality control qPCR data
# - Here we cutoff the data points which are falling out of the lowest standards.
# - And then check whether for each date there are still duplicates.
# - Check the variation between the duplicates as well.
# - Check if difference between duplicates are too big; if yes cutoff.
# - And then check whether for each date there are still duplicates.

# calculate differences between duplicate PCR reactions
qPCR_df_avg <- 
  qPCR_final_data %>% 
  group_by(Content, Exp_date, qPCR_date, Sample, PMA, Spiked, primer) %>%
  summarize(diff_ct = abs(diff(Cq)),
            mean_ct = mean(Cq))

# plot differences for all data points
ggplot(qPCR_df_avg, aes(x=mean_ct, y=diff_ct, color=factor(Sample))) +
  ylab("difference Ct") +
  xlab("mean Ct") +
  geom_hline(yintercept = 2, color = 'red') +
  geom_point(aes(alpha=0.5)) +
  theme_light() + 
  theme(text = element_text(size=10), 
        axis.text.x = element_text(vjust = 0.5, hjust=1))

# split for primers; Standards missing now
variance_plot <- 
  qPCR_df_avg %>% 
  ggplot(aes(x=mean_ct, y=diff_ct, color=factor(Sample), shape = factor(Exp_date))) +
  scale_x_continuous(breaks = seq(5, 35, 2)) +
  ylab("difference Ct") +
  xlab("mean Ct") +
  geom_hline(yintercept = 1, color = 'red') +
  geom_point(aes(alpha=0.5)) +
  facet_wrap(~primer) +
  theme_light() + 
  theme(text = element_text(size=10), 
        axis.text.x = element_text(vjust = 0.5, hjust=1)) +
  scale_color_discrete(name = "Sample") +
  scale_alpha(guide = "none") +
  scale_shape_discrete(name = "Date")

variance_plot

ggsave("figures/ELLY_DPOL_ExpIII_variance_plot_between_sample_duplicates.pdf", variance_plot, width = 8, height = 4)


# Check for differences between the duplicates measurements in qPCR.
# Variance in Ct increases with Ct value. 
# In spiked effluent also higher variance observed at low Ct, likely due to .
# For one of the samples using PA_marta the Ct is >1-3.
# - SD Cutoff at <1?
# - Ct cutoff for PA < 32.627460 (highest Ct of Standard)

#calculate LoD
mean_NTC_ct = 
  qPCR_final_data %>% filter(Content == "NTC-01") %>%
  filter(Cq > 1) %>%
  mutate(mean_Cq = mean(Cq.Mean),
         sd_Cq = sd(Cq.Mean))
sd_NTC_ct = mean_NTC_ct$sd_Cq
mean_NTC_ct  = mean_NTC_ct$mean_Cq

LOD = mean_NTC_ct[1] + (3 * sd_NTC_ct[1]) #29.93
# according to formula CT cutoff set at 29.93, but taking the lowest Cq of blank gives 29.79

# verwijderd ook de blanks
qPCR_final_data_clean <- 
  qPCR_final_data %>% 
  filter(!(primer == "PA_marta" & Cq > 32.6)) %>% 
  filter(Cq < 29.79716)

qPCR_final_data %>% 
filter(!(primer == "PA_marta" & Cq > 32.6)) %>% 
filter(!Cq < 29.79716) %>% 
group_by(qPCR_date, Sample, Content, Sample_type) %>% 
  filter(Sample_type == "effluent")

# average and cutoff SD and remove non duplicates
qPCR_df_avg_clean <- 
  qPCR_final_data_clean %>% 
  group_by(Content, Exp_date, qPCR_date, Sample, PMA, Spiked, primer) %>%
  summarize(diff_ct = abs(diff(Cq)),
            mean_ct = mean(Cq)) %>%
  filter(diff_ct<1) %>%
  mutate(Exp_date = ifelse(Sample == "Standard", "Standard", Exp_date))

# Check again after QC
# shape on exp_date; Standards missing now
ggplot(qPCR_df_avg_clean, aes(x=mean_ct, y=diff_ct, color=factor(Sample))) +
  scale_x_continuous(breaks = seq(5, 35, 2)) +
  ylab("difference Ct") +
  xlab("mean Ct") +
  geom_hline(yintercept = 1, color = 'red') +
  geom_point(aes(alpha=0.5)) +
  facet_wrap(~primer) +
  theme_light() + 
  theme(text = element_text(size=10), 
        axis.text.x = element_text(vjust = 0.5, hjust=0.5)) 

```

After standard curve QC, regression analysis was applied to assess proportion of explained variance for each primer set's standard curve, and a scatter plot was generated.

```{r standard_curve statistics and plot, warning=F, message=F}

# take mean Cq (of technical duplicates) and Log starting quantity for standards
qPCR_df_avg <- 
  qPCR_final_data_clean %>%
  group_by(Content, Sample, qPCR_date, Exp_date, PMA, 
           sample_volume_ul, Spiked, primer, Log.Starting.Quantity) %>%
  summarise(mean_Cq = mean(as.numeric(Cq)), 
            mean_Log.Starting.Quantity = mean(as.numeric(Log.Starting.Quantity)),
            .groups = "drop") %>%
  filter(Sample != "Blank" & n() >= 2) # only keep samples that for which 2 duplicates have been retained

# plot standard curve data for exploration
standard_plot <- 
  qPCR_df_avg %>% 
  filter(Sample == "Standard") %>% 
  ggplot(aes(x= mean_Cq, y = mean_Log.Starting.Quantity, color=Sample)) +
  geom_smooth(method = "lm", se = F, color = "grey70", alpha=0.5) +
  geom_point(aes(color=Content)) +
  facet_wrap(~primer) +
  theme_light() +
  scale_x_continuous(breaks = seq(0, 30, by = 2)) +
  scale_y_continuous(breaks = seq(0, 9, by = 1))

# view standard curves
standard_plot 

# save standard curves
ggsave("figures/ELLY_DPOL_ExpIII_qPCR_standard_curve_plot.pdf", standard_plot, width = 8, height = 4)


# plot all sample data (means of duplicates) on top of standard curves
datapoints_raw <- 
  qPCR_df_avg %>% 
  ggplot(aes(x= mean_Cq, y = mean_Log.Starting.Quantity, color=Sample, alpha = 0.5)) +
    geom_point(aes(color=Sample, shape = PMA)) +
  facet_wrap(~primer) +
  theme_light() +
  scale_x_continuous(breaks = seq(0, 30, by = 2)) +
  scale_y_continuous(breaks = seq(0, 9, by = 1))

# view standard curve and sample data collated
datapoints_raw

# save standard curve and sample data collated
ggsave("figures/ELLY_DPOL_ExpIII_qPCR_standard_curve_and_datapoints_raw.pdf", datapoints_raw, width = 8, height = 4)


# calculate Pearson r for each standard curves
  qPCR_df_avg %>% 
  filter(Sample == "Standard") %>% 
  mutate(primer = factor(primer)) %>% 
  nest(data = -primer) %>% 
  mutate(pearson_r = map(data, ~cor.test(.x$mean_Cq, .x$Log.Starting.Quantity)),
         cor = map(pearson_r, tidy)
        ) %>% 
  unnest(cor)

# calculate linear regressions for standard curve of each primer set to obtain R2
stdcurve_regressions = 
  qPCR_df_avg %>% 
  filter(Sample == "Standard") %>% 
  mutate(primer = factor(primer)) %>% 
  nest(data = -primer) %>% 
  mutate(
    fit = map(data, ~ lm(mean_Cq ~ mean_Log.Starting.Quantity, data = .x)),
    tidied = map(fit, tidy),
    glanced = map(fit, glance),
    augmented = map(fit, augment)
  )

# view primer specific R2
stdcurve_regressions %>% 
  unnest(glanced) %>% 
  dplyr::select(primer, fit, r.squared, df,df.residual, p.value) %>% 
  mutate(p.val_format = format(p.value, scientific = T))

# save primer specific R2
stdcurve_regressions %>% 
  unnest(glanced) %>% 
  dplyr::select(primer, fit, r.squared, "F.statistic" = statistic, df,df.residual, p.value) %>% 
  mutate(p.val_format = format(p.value, scientific = T)) %>% 
  write_excel_csv("output_data/ELLY_DPOL_ExpIII_qPCR_standard_curve_statistics.csv", col_names = T)
  

# model run and slope estimates of interaction primer x Log quantity
model_lm_stdcurve = 
  qPCR_df_avg %>% 
  filter(Sample == "Standard") %>% 
  mutate(primer = factor(primer)) %>% 
  with(lm(mean_Cq ~ mean_Log.Starting.Quantity*primer))

# extract std curve slopes
std_curve_slopes = 
emtrends(model_lm_stdcurve, ~ primer, var = "mean_Log.Starting.Quantity")

# function to calculate PCR efficiencies
calculate_efficiencies <- function(slopes) {
  # Calculate efficiencies for each slope
  efficiencies <- (10^(-1 / slopes) - 1) * 100
  return(efficiencies)
}

# extract all slopes
slopes_standards = testInteractions(model = model_lm_stdcurve, 
                   fixed = "primer", slope = "mean_Log.Starting.Quantity")
intercepts_lm_stdcurve = coef(model_lm_stdcurve)

# calculate PCR efficiencies
efficiencies = 
std_curve_slopes %>% 
  broom::tidy() %>% as_tibble() %>% 
  rowwise() %>% 
  mutate(PCR_efficiency = calculate_efficiencies(mean_Log.Starting.Quantity.trend)) %>% 
  dplyr::select(primer, PCR_efficiency) %>% 
  ungroup()
efficiencies
# extract primer-specific intercepts
intercept_bac_16S = intercepts_lm_stdcurve[["(Intercept)"]] 
intercept_PA_marta = intercepts_lm_stdcurve[["(Intercept)"]] + intercepts_lm_stdcurve[["primerPA_marta"]]
intercept_ybbw = intercepts_lm_stdcurve[["(Intercept)"]] + intercepts_lm_stdcurve[["primerybbw"]]

# extract primer specific slopes
slope_16S <- slopes_standards$Value[1]
slope_PA_marta <- slopes_standards$Value[2]
slope_ybbw <- slopes_standards$Value[3]

# Calculate Ct0 using intercept and slope
Ct0_16S <- -intercept_bac_16S / slope_16S
Ct0_PA_marta <- -intercept_PA_marta / slope_PA_marta
Ct0_ybbw <- -intercept_ybbw / slope_ybbw

```

Estimation of log target copy concentrations for samples was based on the standard curves. PCR efficiencies based on the standard curves were `r round(efficiencies$PCR_efficiency[1], digits=0)`% for universal 16S rRNA targeting primers, `r round(efficiencies$PCR_efficiency[2], digits=0)`% for in-house developed *Pseudomonas* primers, and `r round(efficiencies$PCR_efficiency[3], digits=0)`% for the *E.coli* specific primers developed by Walker *et al.* (2017).<br>
To assess survival of ozonation and therein the effect of PMA on survival estimates, spiked effluent and ozonated samples have been compared.

```{r normalization and analysis of qPCR data, warning=F, message=F, fig.height=6, fig.width=9}

# correcting for gene copies (rrnDB database accessed 15 June 2022)
copies_16S_per_EC_cell = 7
copies_16S_per_PA_cell = 4
copies_ybbw_per_EC_cell = 1
copies_16s_on_average = 4 # average taken of rrnDB

# calculate copies per mL in original sample based on Cq or Ct values
qPCR_df_curve_corrected <- 
  qPCR_df_avg %>%
  filter(Sample != "Standard") %>% # deselect standard curve data points
  mutate(ax = case_when(primer == "bac_16S" ~ mean_Cq-intercept_bac_16S,
                        primer == "PA_marta" ~ mean_Cq-intercept_PA_marta,
                        primer == "ybbw" ~ mean_Cq-intercept_ybbw,
                        TRUE ~ NA_real_)) %>% 
  mutate(log_copies_in_reaction_well = 
              case_when(primer == "bac_16S" ~ abs(ax)/abs(slope_16S),
                        primer == "PA_marta" ~ abs(ax)/abs(slope_PA_marta),
                        primer == "ybbw" ~ abs(ax)/abs(slope_ybbw),
                        TRUE ~ NA_real_)) %>%
  mutate(log_copies_per_well_adjusted  = log_copies_in_reaction_well + 2.4) %>% # +1 log for 10x dilution factor DNA extract in PCR and plus 1.4 for 2 ul DNA in reaction from 50 ul eluted DNA from the total sample
  mutate(copies_per_DNA_extract  = 10^log_copies_per_well_adjusted) %>%
  mutate(copies_mL_original_sample = copies_per_DNA_extract/sample_volume_ul*1000) %>%
  mutate(cells_per_mL_sample = 
              case_when(primer == "bac_16S" & Spiked == "none" ~ copies_mL_original_sample/copies_16s_on_average,
                        primer == "bac_16S" & Spiked == "Ecoli" ~ copies_mL_original_sample/copies_16S_per_EC_cell,
                        primer == "bac_16S" & Spiked == "Paeruginosa"  ~ copies_mL_original_sample/copies_16S_per_PA_cell,
                        primer == "PA_marta" ~ copies_mL_original_sample/copies_16S_per_PA_cell,
                        primer == "ybbw" ~ copies_mL_original_sample/copies_ybbw_per_EC_cell,
                        TRUE ~ NA_real_)) %>%
  mutate(Sample = factor(Sample, levels = c("effluent","Ecoli_culture","Paeruginosa_culture","effluent_spiked","ozone"))) 

# summarize estimated quantities
qPCR_df_curve_corrected_avg <- 
  qPCR_df_curve_corrected %>% 
  group_by(Sample, Exp_date, PMA, Spiked,primer) %>% 
  summarise(mean_cells_per_mL_sample = mean(cells_per_mL_sample),
            log_mean_cells_per_mL_sample = log10(mean_cells_per_mL_sample))


# plot all data
removed_qPCR <- 
qPCR_df_curve_corrected_avg %>% 
  filter(grepl("effluent_spiked|ozone",Sample)) %>% 
  mutate(primer = factor(primer, 
                         levels = c("bac_16S", "ybbw", "PA_marta"), 
                         labels = c("Universal bacterial primers", 
                                    "*E. coli* primers",
                                    "*P. aeruginosa* primers")),
         Spiked = factor(Spiked, levels = c("Ecoli", "Paeruginosa"), 
                         labels = c("*Escherichia coli*", "*Pseudomonas aeruginosa*"))) %>%
 ggplot(aes(x = Sample, y = log10(mean_cells_per_mL_sample), color = PMA)) +
  stat_summary(fun.data = "mean_se", geom = "bar", alpha = 0.7, 
             aes(group=PMA, fill=PMA), 
             position = position_dodge(width = 0.7), width=0.5) +
  geom_point(aes(fill=PMA),
           shape = 21, color="white", 
           position=position_jitterdodge(dodge.width = 0.7, jitter.width = 0.1), 
           size=2, alpha = 0.7) +
  ylab(bquote("log"[10] * " 16S rRNA gene copies mL"^-1 *" sample")) +
  xlab("sample types") +
  facet_wrap(~primer+Spiked, scales = "free_x") +
  theme_light() +
  theme(text = element_text(size=10), 
        axis.text.x = element_text(vjust = 0, hjust=0.5),
        strip.text = element_markdown(),
        axis.ticks.x = element_blank(),
        legend.position="bottom") +
  scale_y_continuous(breaks = seq(0, 9, by = 1)) +
  scale_color_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  scale_fill_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) 

# view qPCR differences between spiked experiments after normalization
removed_qPCR

# save qPCR differences between spiked experiments after normalization
ggsave("figures/ELLY_DPOL_ExpIII_removed_qPCR.pdf", removed_qPCR, width = 6, height = 5)




###### compare all sample types

#plot removal
strip.labels = labeller(primer = c(bac_16S = "Universal bacterial primers", 
                                            PA_marta = "*P. aeruginosa* primers", 
                                            ybbw = "*E. coli* primers"))


# plot qPCR data for all but effluent
all_qPCR <- 
qPCR_df_curve_corrected_avg %>% 
  filter(Sample != "effluent") %>% 
  mutate(combi = factor(paste0(Sample, "_", PMA), levels = c("Ecoli_culture_none", "Ecoli_culture_PMA",
                                                             "Paeruginosa_culture_none", "Paeruginosa_culture_PMA",
                                                             "effluent_spiked_none", "effluent_spiked_PMA",
                                                             "ozone_none", "ozone_PMA"),
                        labels = c(" *E.coli*<br>culture", " ", "*P.aerug.*<br> culture", "  ",
                                   "  spiked<br> effluent", "   ", "ozonated<br> effluent", "    "))) %>%
  ggplot(aes(x = combi, y = log_mean_cells_per_mL_sample)) +
  geom_boxplot(aes(fill = PMA),color = "grey80", outlier.shape = NA, alpha=0.5, position = position_dodge(width = 1)) + 
  # geom_point(aes(color=PMA, group = interaction(Exp_date, PMA), shape = Exp_date),
  #            position=position_jitterdodge(dodge.width = 1, jitter.width = 0.3), size=2) +
  lemon::geom_pointpath(aes(color = PMA, shape = Exp_date, group = interaction(Sample, Exp_date)), 
                        position = position_dodge(width = 0.7), linesize = 0.3, size = 2,  linecolor = "grey40", distance = 0.1)  +
  ylab(bquote("log"[10] * " 16S rRNA gene copies mL"^-1 * " sample")) +
  xlab("sample types") +
  facet_wrap(~primer, scales = "free_x",
             labeller = strip.labels) +
  theme_light() +
  theme(text = element_text(size=10), 
        axis.text.x = element_markdown(vjust = 1, angle = 0, hjust = 0.3,  margin = margin(r = -22)), 
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank(),
        axis.ticks.x = element_blank(),
        legend.position="bottom",
        strip.text = element_markdown(size = 8)) +
  scale_y_continuous(breaks = seq(0, 11, by = 1)) +
  scale_shape_discrete(name = "Date") +
  scale_color_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  scale_fill_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) 


# plot qPCR data for effluent only
effluent_qPCR <- 
qPCR_df_curve_corrected_avg %>% 
  filter(Sample == "effluent") %>% 
  mutate(primer = factor(primer, 
                         levels = c("bac_16S", "PA_marta", "ybbw"), 
                         labels = c("Universal<br>bacterial<br>primers", 
                                    "*P. aerug.*<br> primers", 
                                    "*E. coli*<br> primers"))) %>% 
  ggplot(aes(x = primer, y = mean_cells_per_mL_sample)) +
  geom_boxplot(aes(color=PMA),outlier.shape = NA, alpha=0.5, show.legend = F, position = position_dodge(width = 1), width = 0.5) + 
  geom_point(position=position_jitterdodge(dodge.width = 1, jitter.width = 0.3), size=2,
             aes(color=PMA, group = interaction(PMA, Sample), shape = Exp_date), show.legend = F) +
  # lemon::geom_pointpath(aes(color = PMA, shape = Exp_date, group = interaction(primer, Exp_date)), 
  #                       position = position_jitterdodge(dodge.width = 1, jitter.width = 0.1), linesize = NA, size = 2,  
  #                       linecolor = "grey40", distance = 0.1, show.legend = F)  +
  labs(y = bquote("16S rRNA gene copies mL"^-1 * " sample"), 
       x = "Primer set") +
  facet_wrap(~Sample, scales = "free_x") +
  theme_light() +
  theme(text = element_text(size=10), 
        panel.grid.minor = element_blank(),
        axis.title.y = element_text(vjust = 0.5, hjust = 0.5),
        panel.grid.major.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.x = element_markdown(vjust = 0.5, hjust=0.5, angle = 0, margin = margin(r = -5)), 
        legend.position="bottom",
        strip.text = element_markdown(size = 8)) +
  scale_shape_discrete(name = "Date") +
  scale_color_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  scale_fill_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) 


# plot effluent and log10 concentration data 
plot_combi = 
patchwork::wrap_plots(effluent_qPCR,
  all_qPCR) + patchwork::plot_layout(#nrow = 2, 
                         ncol = 2,
                         widths = c(1, 4),
                         #heights = c(2,2),
                         guides = "collect"
                         )  & theme(legend.position = "bottom")

# view combi plot
plot_combi

# save combi plot
ggsave("figures/ELLY_DPOL_ExpIII_combi_plot.pdf", plot_combi, width = 9, height = 4)

```


```{r ExpIII_plot removal efficiency bar, warning=F, message=F}

qPCR_df_curve_corrected_avg %>% 
  filter(Sample %in% c("effluent_spiked","ozone")) %>% 
  dplyr::select(-mean_cells_per_mL_sample) %>% 
  pivot_wider(names_from = "Sample", values_from = "log_mean_cells_per_mL_sample") %>% 
  filter(primer != "bac_16S") %>% 
  group_by(Spiked, PMA, primer) %>% 
  summarise(eff_mean_log_cells_ml_sample = mean(effluent_spiked),
            O3_mean_log_cells_ml_sample = mean(ozone))

# calculate reductions caused by ozonation for PMA and no-PMA treated samples for each primer set
qPCR_df_reduction <- 
qPCR_df_curve_corrected_avg %>% 
  filter(Sample %in% c("effluent_spiked","ozone")) %>% 
  dplyr::select(-mean_cells_per_mL_sample) %>% 
  pivot_wider(names_from = "Sample", values_from = "log_mean_cells_per_mL_sample") %>% 
  mutate(log_reduction_qPCR = effluent_spiked - ozone,
         primer = factor(primer, 
                         levels = c("bac_16S", "ybbw", "PA_marta"), 
                         labels = c("Universal<br>bacterial primers", 
                                    "*E. coli* primers",
                                    "*P. aeruginosa* primers")),
         Spiked = factor(Spiked,
                         levels = c("Ecoli", "Paeruginosa", "none"),
                         labels = c("*E. coli*", "*P. aeruginosa*", "none")))

# plot reductions per PMA treatment and primer set
qPCR_reduction <- 
  qPCR_df_reduction %>% 
  ggplot(aes(x = Spiked, y = log_reduction_qPCR)) +
  geom_boxplot(aes(fill = PMA),color = "grey80", width = 0.5, outlier.shape = NA, alpha=0.5, position = position_dodge(width = 0.7)) + 
  geom_point(position=position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
             size=2, aes(group=factor(PMA), color = factor(PMA), shape = factor(Exp_date))) +
  facet_wrap(~primer, scales = "free_x") +
  theme_light() +
  theme(strip.text = element_markdown(size = 8),
        axis.title.y = element_markdown(),
        axis.text.x = element_markdown(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank()) +
  scale_y_continuous(breaks = c(0.5, 1, 1.5, 2, 2.5)) +
  scale_shape_discrete(name = "Date") +
  scale_color_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
    scale_fill_manual(name = "Treatment",
                     values = c(pal[7], pal[6]), 
                     labels = c("no PMA", "PMA")) +
  labs(y = "Concentration reduction <br> <span>&Delta;</span>log<sub>10</sub> copies mL<sup>-1</sup> sample", 
       x = "Spiked species") 

# view reduction plot
qPCR_reduction = qPCR_reduction & theme(legend.position = "bottom")
qPCR_reduction
  # save reduction plot
ggsave("figures/ELLY_DPOL_ExpIII_qPCR_reduction.pdf", qPCR_reduction, width = 8, height = 4)

```


```{r ExpIII_plot reductions across methods, warning=F, message=F}

# make one df with date, species, method type and log reduction.
all_qPCR_reduction_df <- 
  qPCR_df_reduction %>%
  filter(primer %in% c('*E. coli* primers', '*P. aeruginosa* primers')) %>%  # Using %in% to filter for multiple values in primer
  rename(exp_date = Exp_date, species = Spiked, reduction = log_reduction_qPCR) %>% 
  group_by(exp_date, species, PMA) %>%
  dplyr::select(exp_date, PMA, species, reduction) %>%
  mutate(exp_date = as.numeric(exp_date),
         method = ifelse(PMA == "none", "qPCR", "PMA")) %>% # Correcting mutate statement
  ungroup() %>%
  dplyr::select(-PMA)

culture_reduction <- 
  complete_culturing_df %>% 
  dplyr::select(exp_date, species, sample_type, mean_CFU_mL) %>%
  pivot_wider(names_from = sample_type, values_from = mean_CFU_mL, values_fill = NA_real_) %>%
  mutate(log_mean_CFU_mL.Es = log10(effluent_spike),
    log_mean_CFU_mL.O3 = log10(ozone),
    reduction_log_CFU_mL = log_mean_CFU_mL.Es - log_mean_CFU_mL.O3)
  
culture_reduction2 <- 
  culture_reduction %>%
  dplyr::select(exp_date, species, reduction_log_CFU_mL) %>%
  rename(reduction = reduction_log_CFU_mL) %>%
  mutate(method = "culturing")

all_reduction_df <- 
  bind_rows(all_qPCR_reduction_df, 
            culture_reduction2) %>% 
  mutate(method = ifelse(method == "PMA", "PMA-qPCR", method),
         species = ifelse(species == "Ecoli", "*E. coli*", species),
         species = ifelse(species == "Paeruginosa", "*P. aeruginosa*", species),
         species = ifelse(species == "E.coli", "*E. coli*", species),
         species = ifelse(species == "P.aeruginosa", "*P. aeruginosa*", species))

#plots
reduction_method_plot <- 
  all_reduction_df %>% 
  ggplot(aes(x = method, y = reduction)) +
  geom_boxplot(aes(fill = method),color = "grey80", width = 0.5, outlier.shape = NA, alpha=0.5, position = position_dodge(width = 0.7)) + 
  geom_point(position=position_jitterdodge(dodge.width = 0.5), size=2, aes(shape = factor(exp_date), color = factor(method))) +
  facet_wrap(~species) +
  theme_light() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        strip.text = element_markdown(size = 10),
        axis.title.y = element_markdown(size = 12),
        axis.title.x = element_markdown(size = 12),
        legend.position = "right", 
        legend.direction = "vertical",
        panel.grid.minor = element_blank(),
        panel.grid.major.x = element_blank()) +
  labs(x = "Quantification method", 
       y = "Concentration reduction <br> <span>&Delta;</span>log<sub>10</sub> cells mL<sup>-1</sup> sample"
       #title = "Reduction of spiked E. coli or P. aeurinogsa in WWTP effluent after ozonation"
       ) +
  scale_y_continuous(breaks = seq(0, 5.5, by = 0.5)) +
  scale_shape_discrete(name = "Date") +
  scale_color_manual(name = "Method", values = c("#EFA00B", "#2E86AB", "#F56476")) +
  scale_fill_manual(name = "Method", values = c("#EFA00B", "#2E86AB", "#F56476"))

  
# view reduction plot comparing methods
reduction_method_plot

# save reduction plot comparing methods
ggsave("figures/ELLY_DPOL_ExpIII_all_methods_log_reduction.pdf", reduction_method_plot, width = 8, height = 4)

```

```{r ExpIII_stats reductions across methods, warning=F, message=F}

# build simple LM model
lm_reduction = lm(reduction ~ method*species, all_reduction_df)
lm_reduction


# view ANOVA
anova_lmm_reduction =
anova(lm_reduction) %>% 
  broom::tidy() %>% 
  rename("F.statistic" = statistic) %>% 
  mutate(p.value.format = format(p.value, scientific=T),
         sig = 
           case_when(p.value < 0.0001 ~ "****",
                     p.value < 0.001 ~ "***",
                     p.value < 0.01 ~ "**",
                     p.value < 0.05 ~ "*",
                     TRUE ~ ""))

anova_lmm_reduction %>% 
  write.xlsx(file = "output_data/ELLY_DPOL_ExpIII_anova_log_reductions_between_methods.xlsx", overwrite = T)

emmeans_red = emmeans(lm_reduction, ~method*species)
emmeans_red

pairs_red = pairs(emmeans_red, by = "species")
pairs_red


if(!dir.exists("tables")){dir.create("tables")}
pairs_red %>% 
  write.xlsx(file = "output_data/ELLY_DPOL_ExpIII_contrasts_log_reductions_between_methods.xlsx", overwrite = T)

pairs_species = pairs(emmeans_red, by = "method")
pairs_species

```


```{r sessionInfo, warning=F, message=F}

sessionInfo()

```

